Microsoft.OData.AzureFunctions/ODataValueProvider.cs

```diff
@@ -27,15 +25,16 @@ public async Task<object> GetValueAsync()
                 IEdmModel model = this.modelProvider.GetEdmModel();
 
                 // TODO: Add routePrefix to constructor to be passed from the function
-                return ODataBindingHelper.BuildODataQueryOptions(this.request, model, this.clrType, this.routePrefix);
+                Type clrType = typeof(T).GenericTypeArguments.First();
+                return Task.FromResult<object>(ODataBindingHelper.BuildODataQueryOptions(this.request, model, clrType, this.routePrefix));
             }
-            catch (Exception ex)
+            catch (Exception)
```

Comment:
Maybe we should remove the try...catch block altogether since all we're doing is catching the exception and rethrowing it...

Reply:
Or throw an ODataException?